src/main/scala/com/wavesplatform/database/Keys.scala

```diff
@@ -0,0 +1,116 @@
+  def score(height: Int): Key[BigInt] = Key(h(2, height), Option(_).fold(BigInt(0))(BigInt(_)), _.toByteArray)
+
+  private def blockAtHeight(height: Int) = h(3, height)
+
+  def blockAt(height: Int): Key[Option[Block]]          = Key.opt[Block](blockAtHeight(height), Block.parseBytes(_).get, _.bytes())
+  def blockBytes(height: Int): Key[Option[Array[Byte]]] = Key.opt[Array[Byte]](blockAtHeight(height), identity, identity)
+  def blockHeader(height: Int): Key[Option[(BlockHeader, Int)]] =
+    Key.opt[(BlockHeader, Int)](blockAtHeight(height), b => (BlockHeader.parseBytes(b).get._1, b.length), unsupported("Can't write block headers")) // this dummy encoder is never used: we only store blocks, not block headers
+
+  def heightOf(blockId: ByteStr): Key[Option[Int]] = Key.opt[Int](hash(4, blockId), Ints.fromByteArray, Ints.toByteArray)
+
+  def wavesBalanceHistory(addressId: BigInt): Key[Seq[Int]] = historyKey(5, addressId.toByteArray)
+  def wavesBalance(height: Int, addressId: BigInt): Key[Long] =
+    Key(hAddr(6, height, addressId), Option(_).fold(0L)(Longs.fromByteArray), Longs.toByteArray)
+
+  def assetList(addressId: BigInt): Key[Set[ByteStr]]                         = Key(addr(7, addressId), readTxIds(_).toSet, assets => writeTxIds(assets.toSeq))
+  def assetBalanceHistory(addressId: BigInt, assetId: ByteStr): Key[Seq[Int]] = historyKey(8, addressId.toByteArray ++ assetId.arr)
+  def assetBalance(height: Int, addressId: BigInt, assetId: ByteStr): Key[Long] =
+    Key(hBytes(9, height, addressId.toByteArray ++ assetId.arr), Option(_).fold(0L)(Longs.fromByteArray), Longs.toByteArray)
+
+  def assetInfoHistory(assetId: ByteStr): Key[Seq[Int]]        = historyKey(10, assetId.arr)
+  def assetInfo(height: Int, assetId: ByteStr): Key[AssetInfo] = Key(hBytes(11, height, assetId.arr), readAssetInfo, writeAssetInfo)
+
+  def leaseBalanceHistory(addressId: BigInt): Key[Seq[Int]] = historyKey(12, addressId.toByteArray)
+  def leaseBalance(height: Int, addressId: BigInt): Key[LeaseBalance] =
+    Key(hBytes(13, height, addressId.toByteArray), readLeaseBalance, writeLeaseBalance)
+  def leaseStatusHistory(leaseId: ByteStr): Key[Seq[Int]] = historyKey(14, leaseId.arr)
+  def leaseStatus(height: Int, leaseId: ByteStr): Key[Boolean] =
+    Key(hBytes(15, height, leaseId.arr), _(0) == 1, active => Array[Byte](if (active) 1 else 0))
+
+  def filledVolumeAndFeeHistory(orderId: ByteStr): Key[Seq[Int]]           = historyKey(16, orderId.arr)
+  def filledVolumeAndFee(height: Int, orderId: ByteStr): Key[VolumeAndFee] = Key(hBytes(17, height, orderId.arr), readVolumeAndFee, writeVolumeAndFee)
+
+  def transactionInfo(txId: ByteStr): Key[Option[(Int, Transaction)]] = Key.opt(hash(18, txId), readTransactionInfo, writeTransactionInfo)
+  def transactionHeight(txId: ByteStr): Key[Option[Int]] =
+    Key.opt(hash(18, txId), readTransactionHeight, _ => throw new UnsupportedOperationException("Can't write transaction height only"))
```

Comment:
`transactionHeight` uses the same prefix because simply parses the first four bytes from the blob, ignoring transaction bytes (just like `blockHeader`).